src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -1129,14 +1136,19 @@ def execute_post_deploy_hooks(
         package_name: str,
         package_warehouse: Optional[str],
     ):
-        with cls.use_package_warehouse(package_warehouse):
-            execute_post_deploy_hooks(
-                console=console,
-                project_root=project_root,
-                post_deploy_hooks=post_deploy_hooks,
-                deployed_object_type="application package",
-                database_name=package_name,
-            )
+        get_cli_context().metrics.set_counter_default(
+            CLICounterField.POST_DEPLOY_SCRIPTS, 0
+        )
+
+        if post_deploy_hooks:
+            with cls.use_package_warehouse(package_warehouse):
+                execute_post_deploy_hooks(
```

Comment:
Ok I see what you meant. So you don't want to change the invariant that we don't issue a "USE WAREHOUSE" unless at least one script is present. Fair enough for now, but do sync with Parya so it's on her radar too please.